nexus/api.go

```diff
@@ -176,3 +201,107 @@ func addContextTimeoutToHTTPHeader(ctx context.Context, httpHeader http.Header)
 	httpHeader.Set(HeaderRequestTimeout, time.Until(deadline).String())
 	return httpHeader
 }
+
+// Link contains an URL and a Type that can be used to decode the URL.
+// Links can contain any arbitrary information as a percent-encoded URL.
+// It can be used to pass information about the caller to the handler, or vice-versa.
+type Link struct {
+	// URL information about the link.
+	// It must be URL percent-encoded.
+	URL string
+	// Type can describe an actual data type for decoding the URL.
+	// Valid chars: alphanumeric, '_', '.', '/'
+	Type string
+}
+
+const linkTypeKey = "type"
+
+func encodeLink(link Link) (string, error) {
+	if err := validateLinkURL(link.URL); err != nil {
```

Comment:
It would probably be easier for users to provide a URL here and let us serialize it than worry about them properly encoding the values.

Reply:
Well, technically, the URL needs to be correctly encoded to be a valid URL, doesn't it? Eg: `a/b/c` vs `a/b%2Fc`. Here, `%2F` is the encoded value of `/`.